### DISCUSSION TITLE
Is it possible to use a non-string type for a radio value?

### DISCUSSION THREAD

**Discussion by aidan-keay** [adobe/react-spectrum#1939] [Q&A]
I would like to use numbers or objects as a radio value, is this possible? Something like:

```javascript
<RadioGroup<number> onChange={(value: number) => fn(value)}>
  <Radio value={5} />
 <Radio value={8} />
</RadioGroup>
```

**Comment by snowystinger** [comment]
Well, TypeScript would complain, as value is defined as `value: string` matching https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio#value
And this code path would be completely untested.
I'd just make the string coercion.

**Reply by aidan-keay** [reply]
Ok, thanks, yeah I just wanted to check if there was any workaround solution. I will try sticking to strings.
